feat(manager/cargo): support private registry index env var (#20379)
renovatebot/renovate: docs/usage/rust.md

1. Renovate searches in each repository for any `Cargo.toml` files
1. Renovate extracts existing dependencies from `[dependencies]`, `[dev-dependencies]`, `[build-dependencies]` and `[workspace.dependencies]`

> 1. Renovate looks up Cargo configuration to discover index URLs for private registries.

Reviewer: ```suggestion
1. Renovate looks up Cargo configuration to find index URLs for private registries
```

We don't use punctuation at the end of a list item. I like `find` better than `discover`.

1. Renovate resolves the dependency's version using the crates.io API or by cloning the index URL
1. If Renovate finds an update, Renovate will use `cargo update` to update both `Cargo.toml` and `Cargo.lock`

## Enabling Rust Modules Updating

Renovate updates Rust crates by default.

## Cargo configuration and private registry discovery

> Renovate can determine private registry URLs from a `.cargo/config.toml` (or the legacy `.cargo/config`) Cargo configuration file.

Reviewer: ```suggestion
Renovate can find private registry URLs in these Cargo configuration files:

- `.cargo/config.toml`
- `.cargo/config` (legacy)
```

Changes:

- Put config files in a list
- Replace `determine` with the simple word `find`
- Rewrite the sentence a little bit

> Renovate can also detect private registry URLs via a `CARGO_REGISTRIES_<name>_INDEX` [environment variable](https://doc.rust-lang.org/cargo/reference/environment-variables.html#configuration-environment-variables).

Reviewer: ```suggestion
Renovate can also find private registry URLs via a `CARGO_REGISTRIES_<name>_INDEX` environment variable.
Read the [Rust environment variables docs](https://doc.rust-lang.org/cargo/reference/environment-variables.html#configuration-environment-variables) to learn more.
```

How about this:

- replace `detect` with the simple word `find`
- Split the sentence into two parts
- Use a better name for the link to the Rust docs for the environment variable

1. Renovate searches in each repository for any `Cargo.toml` files
1. Renovate extracts existing dependencies from `[dependencies]`, `[dev-dependencies]`, `[build-dependencies]` and `[workspace.dependencies]`
1. Renovate looks up Cargo configuration to find index URLs for private registries
1. Renovate resolves the dependency's version using the crates.io API or by cloning the index URL
1. If Renovate finds an update, Renovate will use `cargo update` to update both `Cargo.toml` and `Cargo.lock`

## Enabling Rust Modules Updating

Renovate updates Rust crates by default.

## Cargo configuration and private registry discovery

Renovate can find private registry URLs in these Cargo configuration files:

- `.cargo/config.toml`
- `.cargo/config` (legacy)

> Renovate can also find private registry URLs via a `CARGO_REGISTRIES_<name>_INDEX` environment variable.

Reviewer: ```suggestion

Renovate can also find private registry URLs via a `CARGO_REGISTRIES_<name>_INDEX` environment variable.
```
